extensions/replication/queueProcessor/task.js

```diff
@@ -20,17 +24,96 @@ const log = new werelogs.Logger('Backbeat:QueueProcessor:task');
 werelogs.configure({ level: config.log.logLevel,
     dump: config.log.dumpLevel });
 
-const activeQProcessors = {};
+function getCRRStatusZkPath() {
+    return `${zookeeperReplicationNamespace}${ZK_CRR_STATUS_PATH}`;
+}
+
+/**
+ * On startup and when replication sites change, create necessary zookeeper
+ * status node to save persistent state.
+ * @param {node-zookeeper-client.Client} zkClient - zookeeper client
+ * @param {String} site - replication site name
+ * @param {Function} done - callback(error, status) where status is a boolean
+ * @return {undefined}
+ */
+function setupZkStatusNode(zkClient, site, done) {
+    const path = `${getCRRStatusZkPath()}/${site}`;
+    zkClient.create(path, Buffer.from('true'), err => {
+        if (err && err.name !== 'NODE_EXISTS') {
+            log.fatal('could not setup zookeeper node', {
+                method: 'QueueProcessor:task',
+                zookeeperPath: path,
+                error: err,
+            });
+            return done(err);
+        }
+        if (err && err.name === 'NODE_EXISTS') {
```

Comment:
For readability and avoiding extra unnecessary condition checks, I would recommend to change the structure to:

```js
if (err && err.name === 'NODE_EXISTS') {
    ...
    return done();
}
if (err) {
    return done(err);
}
...
```